hail/src/test/scala/is/hail/expr/ir/IRSuite.scala

```diff
@@ -2502,13 +2488,13 @@ class IRSuite extends HailSuite {
 
   @DataProvider(name = "compareDifferentTypes")
   def compareDifferentTypesData(): Array[Array[Any]] = Array(
-    Array(FastIndexedSeq(0.0, 0.0), TArray(+TFloat64()), TArray(TFloat64())),
-    Array(Set(0, 1), TSet(+TInt32()), TSet(TInt32())),
-    Array(Map(0L -> 5, 3L -> 20), TDict(+TInt64(), TInt32()), TDict(TInt64(), +TInt32())),
-    Array(Interval(1, 2, includesStart = false, includesEnd = true), TInterval(+TInt32()), TInterval(TInt32())),
-    Array(Row("foo", 0.0), TStruct("a" -> +TString(), "b" -> +TFloat64()), TStruct("a" -> TString(), "b" -> TFloat64())),
-    Array(Row("foo", 0.0), TTuple(TString(), +TFloat64()), TTuple(+TString(), +TFloat64())),
-    Array(Row(FastIndexedSeq("foo"), 0.0), TTuple(+TArray(TString()), +TFloat64()), TTuple(TArray(+TString()), +TFloat64()))
+    Array(FastIndexedSeq(0.0, 0.0), Array(TFloat64()), TArray(TFloat64())),
```

Comment:
note that I'll make a followup PR to restore this test to actually checking comparisons of different ptypes -- but it already was testing PType.canonical against PType.canonical.

Reply:
sure